File: python/5/escola.py
class Escola:
    def __init__(self, nome):
        self.nome = nome
        self.professores = []

    def adicionar_professor(self, professor):
        if professor not in self.professores:
            self.professores.append(professor)
            professor.adicionar_escola(self)

    def remover_professor(self, professor):
        if professor in self.professores:
            self.professores.remove(professor)
            professor.remover_escola(self) 
    def __str__(self):
        return f"Escola: {self.nome}"


class Professor:
    def __init__(self, nome):
        self.nome = nome
        self.escolas = []

    def adicionar_escola(self, escola):
        if escola not in self.escolas:
            self.escolas.append(escola)
            escola.adicionar_professor(self)

    def remover_escola(self, escola):
        if escola in self.escolas:
            self.escolas.remove(escola)
            escola.remover_professor(self)

    def __str__(self):
        return f"Professor: {self.nome}"

File: java/Escola.java
import java.util.ArrayList;
import java.util.List;

public class Escola {
    private String nome;
    private List<Professor> professores;

    public Escola(String nome) {
        this.nome = nome;
        this.professores = new ArrayList<>();
    }

    public String getNome() {
        return nome;
    }

    public List<Professor> getProfessores() {
        return professores;
    }

    public void adicionarProfessor(Professor professor) {
        if (!professores.contains(professor)) {
            professores.add(professor);
            professor.adicionarEscola(this); // Associação bidirecional
        }
    }

    public void removerProfessor(Professor professor) {
        if (professores.contains(professor)) {
            professores.remove(professor);
            professor.removerEscola(this); // Remoção bidirecional
        }
    }

    @Override
    public String toString() {
        return "Escola: " + nome;
    }
}

